caas/kubernetes/clientconfig/k8s.go

```diff
@@ -132,21 +132,18 @@ func contextsFromConfig(config *clientcmdapi.Config) (map[string]Context, error)
 }
 
 func cloudsFromConfig(config *clientcmdapi.Config, cloudName string) (map[string]CloudConfig, error) {
-
 	clusterToCloud := func(cluster *clientcmdapi.Cluster) (CloudConfig, error) {
 		attrs := map[string]interface{}{}
 
-		// TODO(axw) if the CA cert is specified by path, then we
-		// should just store the path in the cloud definition, and
-		// rely on cloud finalization to read it at time of use.
+		k8sCAData := cluster.CertificateAuthorityData
```

Comment:
Is this something we can put into a unit test that having CertificateAuthority is supported but doesn't modify CertificateAuthorityData?

Given the other error message, doesn't CertificateAuthorityData supercede CertificateAuthority? This code looks like it overwrites Data if CertificateAuthority is set.

The error that we got was
```
error: certificate-authority-data and certificate-authority are both specified for kubernetes. certificate-authority-data will override.
```

Which would hint that -data is primary, except it is an error so it just refuses.
(Note this also happens if you just kubectl with both available according to:
https://stackoverflow.com/questions/52302684/eks-error-certificate-authority-data-and-certificate-authority-are-both-specif
)

Reply:
I've made some changes to cover this.